### PR TITLE
[spotify-api] typings for audio analysis object

### PR DESCRIPTION
**Pull request by eegli**
Typings for the [track audio analysis endpoint](https://developer.spotify.com/documentation/web-api/reference/#/operations/get-audio-analysis).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/spotify-api/index.d.ts [hunk 1]
```diff
@@ -985,6 +982,90 @@ declare namespace SpotifyApi {
...
+    interface AudioAnalysisObject {
+        meta: {
+            analyzer_version: string
+            platform: string,
+            detailed_status: string,
+            status_code: number,
+            timestamp: number,
+            analysis_time: number,
+            input_process: string
+        },
```
Comment (piksel) [inline types/spotify-api/index.d.ts:998]: I think these sub-types ought to have their own interfaces, that way it's easier to use only the relevant parts as a argument type for example.

File: types/spotify-api/index.d.ts [hunk 2]
```diff
@@ -672,7 +669,7 @@ declare namespace SpotifyApi {
      * GET /v1/tracks/{id}
      * https://developer.spotify.com/web-api/get-track/
      */
-    interface SingleTrackResponse extends TrackObjectFull {}
+    interface SingleTrackResponse extends TrackObjectFull { }
```
Comment (piksel) [inline types/spotify-api/index.d.ts:672]: I presume these are auto-formatted? If so, they aren't strictly part of this PR, but including them would make future PRs easier, so I think you can keep them.

**Review by piksel — CHANGES_REQUESTED** [review]
Split the interface into sub interfaces, but otherwise this looks good and matches what I manually generated for this API in a project.

File: types/spotify-api/index.d.ts (hunk 1 above)
Comment (eegli) [reply types/spotify-api/index.d.ts]: Good point. I've extracted the subtypes into their own interfaces. Also, I found out that there is an official prettier configuration for this repo. With the latest commit, the files have been [formatted properly](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

**Review by piksel — APPROVED** [review]

**Comment by eegli** [comment]
Ready to merge

**Pull request merged**